python/python_helper_sympy.py

```diff
@@ -85,116 +122,130 @@ def __init__(self):
+class HasCommentError(BaseSympyError):
+    offset: int
 
 class CheckNumbers(ast.NodeTransformer):
-    def visit_Num(self, node):
+    def visit_Constant(self, node: ast.Constant) -> ast.Constant:
         if isinstance(node.n, int):
-            return ast.Call(func=ast.Name(id='_Integer', ctx=ast.Load()), args=[node], keywords=[])
+            return cast(ast.Constant, ast.Call(func=ast.Name(id='_Integer', ctx=ast.Load()), args=[node], keywords=[]))
         elif isinstance(node.n, float):
             raise HasFloatError(node.col_offset, node.n)
         elif isinstance(node.n, complex):
-            raise HasComplexError(node.col_offset, node.n)
+            raise HasComplexError(node.col_offset, str(node.n))
         return node
 
 class CheckWhiteList(ast.NodeVisitor):
-    def __init__(self, whitelist):
+    def __init__(self, whitelist: ASTWhitelistT) -> None:
         self.whitelist = whitelist
 
-    def visit(self, node):
+    def visit(self, node: Any) -> None:
         if not isinstance(node, self.whitelist):
             node = get_parent_with_location(node)
             raise HasInvalidExpressionError(node.col_offset)
         return super().visit(node)
 
 class CheckFunctions(ast.NodeVisitor):
-    def __init__(self, functions):
+    def __init__(self, functions: SympyMapT) -> None:
         self.functions = functions
-    def visit_Call(self, node):
+
+    def visit_Call(self, node: Any) -> None:
         if isinstance(node.func, ast.Name):
             if node.func.id not in self.functions:
                 node = get_parent_with_location(node)
                 raise HasInvalidFunctionError(node.col_offset, node.func.id)
         self.generic_visit(node)
 
 class CheckVariables(ast.NodeVisitor):
-    def __init__(self, variables):
+    def __init__(self, variables: SympyMapT) -> None:
         self.variables = variables
-    def visit_Name(self, node):
+
+    def visit_Name(self, node: Any) -> None:
         if isinstance(node.ctx, ast.Load):
             if not is_name_of_function(node):
                 if node.id not in self.variables:
                     node = get_parent_with_location(node)
                     raise HasInvalidVariableError(node.col_offset, node.id)
         self.generic_visit(node)
 
-def is_name_of_function(node):
+def is_name_of_function(node: Any) -> bool:
     # The node is the name of a function if all of the following are true:
     # 1) it has type ast.Name
     # 2) its parent has type ast.Call
     # 3) it is not in the list of parent's args
-    return isinstance(node, ast.Name) and isinstance(node.parent, ast.Call) and (node not in node.parent.args)
+    return isinstance(node, ast.Name) and isinstance(node.parent, ast.Call) and (node not in node.parent.args) # type: ignore
 
-def get_parent_with_location(node):
+def get_parent_with_location(node: ast.AST) -> ast.AST:
     if hasattr(node, 'col_offset'):
         return node
-    else:
-        return get_parent_with_location(node.parent)
```

Comment:
seems like an unnecessary change to make

Reply:
We have to change this line anyway to add the type ignore comment.